pkg/fqdn/helpers.go

```diff
@@ -105,12 +105,9 @@ func injectToCIDRSetRules(rule *api.Rule, cache *DNSCache, reMap *regexpmap.Rege
 					allIPs = append(allIPs, ips...)
 				}
 			}
-		}
-
-		// Only overwrite ToCIDRSet for rules with FQDN elements
-		if len(allIPs) > 0 {
 			egressRule.ToCIDRSet = api.IPsToCIDRRules(ip.KeepUniqueIPs(allIPs))
```

Comment:
Maybe leave a comment here stating that IPsToCIDRRules returns a nil slice when there are no IPs.
Did you figure out if there is a difference between ToCIDRSet being nil or an empty slice?